ejb/WFLY-13354-galleon-for-ejb.adoc

```diff
@@ -0,0 +1,110 @@
+(A webservices layer would depend on this.)
+
+2. ejb3. Builds (i.e. depends) on ejb3-lite and adds the discovery subsystem, the MDB pool and the remote connector resource.
+This looks what we provide in standalone-full.xml, except no IIOP integration.
+
+3. ejb3-iiop. Builds (i.e. depends) on *ejb3-lite* and adds the iiop subsystem plus the integration resource in the EJB3 subsystem.
+This *could* depend on *ejb3* instead, but that means EJB is unnecessarily exposed over the HTTP interface and a lot of
+dependencies are brought in for messaging integration.
+
+A user could provision ejb3 + ejb3-iiop and get the equivalent to what's in standalone-full.xml.
+
+There'd be a couple ancillary layers as well:
+
+1. ejb-local-cache. This provides the infinispan subsystem resources related to local ejb caching
+(i.e. the ones in standalone.xml and standalone-full.xml.) The ejb3-lite layer *optionally* depends on this.
+It's an optional dependency because the user when provisioning can *exclude* this layer and instead provision...
+
+2. ejb-dist-cache. This provides the infinispan subsystem resources related to distributed ejb caching
+(i.e. the ones in standalone0ha.xml and standalone-full-ha.xml.)
+
+This pattern of "exclude an optional local caching layer and add a distributed variant" is how web session
+and jpa caching are handled in the layers for those features.
+
+TBH I don't see big FS footprint improvements coming out of these permutations. Big EJB dependencies include (in no particular order.)
+
+* IIOP. But the TM uses IIOP libs so they will be there regardless.
+* Remoting. It would be nice to be able to eliminate the need for remoting, e.g. for ejb3-lite if the user app isn't a remote ejb client. But that doesn't seem trivial and if a server is manageable remoting will be there anyway to support the CLI.
+* JCA stuff. But if there's a datasource it's there anyway.
+* Infinispan. A benefit of a future SLSB-only layer would be this dep could be eliminated. (But it might be used for web caching etc anyway.)
+* Remote artemis integration.  This we do save if ejb3-lite is used.
+
+
+== Test Plan
+
+The test coverage of the Galleon layers added by this proposal is divided in two main groups:
+
+. Test the Galleon layer provision.
+https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]
+will be expanded to test the provision of the added layers when it is used isolated and with other layers.
+For each sort of provision, this test does the following:
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
+. Execution of existing tests on a minimum set of layers. Reuse the existing tests available on the WildFly test suite,
+which are directly testing features added by these layers, and execute them on a server installation provisioned with this layer.
+
```

Comment:
@chengfang can we have some more details here? for example security and clustering test are topics of interest